wormhole-attester/program/src/attest.rs

```diff
@@ -60,9 +65,10 @@ pub const P2W_MAX_BATCH_SIZE: u16 = 5;
 #[derive(FromAccounts)]
 pub struct Attest<'b> {
```

Comment:
The Attest interface has changed. How do plan on achieving 0 downtime for attester cranks?

Reply:
yeah... what's the upgrade process here?

can you make the attestationstatemap optional so that if it's missing, it just sets the time intervals to 0?

Reply:
One possible way to handle it smoothly is making the account optional but idk whether that's possible or not.

The other way is that we run a new attester with the new config that does not crash on failure and keep another one that works with the previous one. The new one should not crash on k8s as there is an exponential backoff on restarting the crashed pod.

Another approach is that attester can handle both types at the same time but i'm not sure that be easy to implement.

Reply:
I was thinking precisely about what Ali said. We run a new attester in parallel with the old one, _before_ deploying new contract version. This new attester will keep crashing intentionally. Then, we perform the program upgrade. This should cause the old attester to start crashing, and the new one to pick up the ball immediately.

Reply:
Everything else should be relatively happy to utilize the new format. the wire format is extended in a backwards compatible way. Any parser that obeys the format should understand that there's trailing bytes it does not need to care for.